### PR TITLE
Rename _scatter_reduce to scatter_reduce and make it unstructured

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #71895
* #71788
* __->__ #71787

Differential Revision: [D33778524](https://our.internmc.facebook.com/intern/diff/D33778524)

Unblocks #68580

### CODE REVIEW

**Comment by mikaylagawarecki** [comment]
@mikaylagawarecki has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33778524).

**Review by cpuhrsch — APPROVED** [review]

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -6010,15 +6010,10 @@
 - func: scatter_add.dimname(Tensor self, Dimname dim, Tensor index, Tensor src) -> Tensor
   variants: function, method
 
-- func: _scatter_reduce(Tensor self, int dim, Tensor index, str reduce, *, int? output_size=None) -> Tensor
-  structured_delegate: _scatter_reduce.out
+- func: scatter_reduce.two(Tensor self, int dim, Tensor index, str reduce, *, int? output_size=None) -> Tensor
```
Comment (bdhirsh) [inline aten/src/ATen/native/native_functions.yaml:6013]: Hey @mikaylagawarecki! It looks like this PR promotes `scatter_reduce` to a public API (you can now call `torch.scatter_reduce` from python), but it doesn't have any public documentation.

For the release, we want to either avoid adding it to python, or adding docs for it. Do you have a preference? If you want to hide it from python, I can make a PR. But if you want to add docs, you'll need to cherry-pick the docs into the release branch (I can point you to the process internally).

cc @albanD

Comment (albanD) [reply aten/src/ATen/native/native_functions.yaml:6013]: Yep that should be documented and cherry picked in the release branch. (or we can revert this for the release if the doc is not ready)

cc @cpuhrsch

Comment (cpuhrsch) [reply aten/src/ATen/native/native_functions.yaml:6013]: Let's make this private again. We didn't land on a conclusion around this function in time. Thanks for checking @bdhirsh @albanD .

**Pull request closed without merging**